### PR TITLE
Show uniques graph contextual to territory selection.

### PR DESCRIPTION
**Pull request by LATaylor-guardian**
- Provides functionality to allow the user to set a target per territory. 
- Makes any target that is being displayed to the user contextual to the territory. If no target is available it will be hidden.

### CODE REVIEW

File: app/controllers/CampaignApi.scala [hunk 1]
```diff
@@ -87,17 +87,22 @@ class CampaignApi(components: ControllerComponents, authAction: AuthAction[AnyCo
     result getOrElse InternalServerError
   }
 
-  def getCampaignPageViewsFromDatalake(id: String) = authAction { _ =>
-    CampaignService.getPageViews(id) match {
+  def getCampaignPageViewsFromDatalake(id: String) = authAction { implicit request =>
+    val territory = Territory(request.getQueryString("territory") getOrElse "global")
```
Comment (kelvin-chappell) [inline app/controllers/CampaignApi.scala]: Could add this as an optional param to the endpoint

File: app/controllers/CampaignApi.scala [hunk 2]
```diff
@@ -87,17 +87,22 @@ class CampaignApi(components: ControllerComponents, authAction: AuthAction[AnyCo
...
       case Left(_)                       => InternalServerError
       case Right(pageViews)              => Ok(Json.toJson(pageViews))
     }
   }
 
   // TODO - fix return types
-  def getCampaignUniquesFromDatalake(id: String) = authAction { _ =>
-    CampaignService.getUniquesDataForGraph(id).map(uniquesData => Ok(Json.toJson(uniquesData))) getOrElse NotFound
+  def getCampaignUniquesFromDatalake(id: String) = authAction { implicit request =>
+    val territory = Territory(request.getQueryString("territory") getOrElse "global")
```
Comment (kelvin-chappell) [inline app/controllers/CampaignApi.scala]: same here

File: app/model/Campaign.scala [hunk 3]
```diff
@@ -18,7 +18,7 @@ case class Campaign(
   startDate: Option[DateTime] = None,
   endDate: DateTime,
   targets: Map[String, Long] = Map.empty,
-  campaignTargets: Option[Map[String, Map[String, Long]]] = None,
+  campaignTargets: Option[Map[String, Map[String, Long]]] = Some(Map.empty),
```
Comment (kelvin-chappell) [inline app/model/Campaign.scala:21]: If this is never none, should it be optional?

**Review by kelvin-chappell — APPROVED** [review]
👍

File: app/model/Campaign.scala (hunk 3 above)
Comment (LATaylor-guardian) [reply app/model/Campaign.scala:21]: Not all campaigns have this field at the moment which causes it to break.

File: app/controllers/CampaignApi.scala (hunk 1 above)
Comment (LATaylor-guardian) [reply app/controllers/CampaignApi.scala]: Yes - will change.

**Pull request merged**